modules/snowflake-loader/src/test/scala/com/snowplowanalytics/snowplow/loader/snowflake/test/PureDAO.scala

```diff
@@ -0,0 +1,47 @@
+package com.snowplowanalytics.snowplow.loader.snowflake.test
+
+import com.snowplowanalytics.snowplow.rdbloader.test.{Pure, PureTransaction, TestState}
+import com.snowplowanalytics.snowplow.loader.snowflake.db.{SfDao, Statement}
+import doobie.Read
+
+case class PureDAO(executeQuery: Statement => Pure[Any], executeUpdate: Statement => Pure[Int])
+
+object PureDAO {
```

Comment:
Would be nice to abstract most of this class out into the core. There is fair bit of duplication between this and redshift version.

Reply:
Same reasoning with this one: https://github.com/snowplow/snowplow-rdb-loader/pull/709#discussion_r803153682
I think current way is cleaner. Another advantage is Doobie doesn't infiltrate other places when we put all statements in one place.